src/main/flight/imu.c

```diff
@@ -196,6 +198,23 @@ static void imuTransformVectorBodyToEarth(t_fp_vector * v)
     v->V.Z = z;
 }
 
+void imuTransformVectorEarthToBody(t_fp_vector_def * v) {
+    const float x = (q0q0 + q1q1 - q2q2 - q3q3) * v->X + 2*(q1q2 + q0q3) * v->Y + 2*(q1q3 - q0q2) * v->Z;
+    const float y = 2*(q1q2 - q0q3) * v->X + (q0q0 - q1q1 + q2q2 - q3q3) * v->Y + 2*(q2q3 + q0q1) * v->Z;
+    const float z = 2*(q1q3 + q0q2) * v->X + 2*(q2q3 - q0q1) * v->Y + (q0q0 - q1q1 - q2q2 + q3q3) * v->Z;
+
+    v->X = x;
+    v->Y = y;
+    v->Z = z;
+}
+
+void imuRebaseEarthToBody(void) {
+    q0 = 1.0;
```

Comment:
This will reset full orientation. It will take considerable time until attitude estimate converges again, which will be quite fun in level mode.
Also if mag is present, it will pull attitude estimate, making headfree mode useless